Add registries settings (#2303)
saucelabs/sauce-docs: docs/web-apps/automated-testing/cypress/yaml/v1.md

---

> ### `registry`

Reviewer: I don't think we should delete it just like that. Because `registries` would only be supported in certain framework + saucectl versions. There's no explanation as to what these are in saucectl or here.

Author: I restored the `registry` field + add a note that this is supported up to <current version>. Then should use `registries`.

<p><small>| OPTIONAL | STRING |</small></p>

:::note

> This setting is supported up to Cypress 12.14.0. For later version, use `registries`.

Reviewer: ```suggestion
This setting is supported up to Cypress 12.14.0. For newer versions, use `registries`.
```

---

### `registries`

<p><small>| OPTIONAL | ARRAY |</small></p>

> Specifies the location of the npm registry, scope and credentials. If the registry is inside a private network, you need to establish a tunnel using [Sauce Connect](/dev/cli/saucectl/#run-tests-on-sauce-labs-with-sauce-connect).

Reviewer 2: ```suggestion
Specifies the location of the npm registry, scope, and credentials. If the registry is inside a private network, you must establish a tunnel using [Sauce Connect](/dev/cli/saucectl/#run-tests-on-sauce-labs-with-sauce-connect).
```

> Specifies the location of the npm registry, scope and credentials. If the registry is inside a private network, you need to establish a tunnel using [Sauce Connect](/dev/cli/saucectl/#run-tests-on-sauce-labs-with-sauce-connect).

Reviewer 2: ```suggestion
Specifies the location of the npm registry, scope, and credentials. If the registry is inside a private network, you must establish a tunnel using [Sauce Connect](/dev/cli/saucectl/#run-tests-on-sauce-labs-with-sauce-connect).
```

> Specifies the location of the npm registry source, scope and credentials. If the registry source is a private address and you are running tests on Sauce Cloud, you can provide access to the registry source using [Sauce Connect](/dev/cli/saucectl/#run-tests-on-sauce-labs-with-sauce-connect).

Reviewer: ```suggestion
Specifies the location of the npm registry, scope and credentials. If the registry is inside a private network, you need to establish a tunnel using [Sauce Connect](/dev/cli/saucectl/#run-tests-on-sauce-labs-with-sauce-connect).
```

```yaml
  registries:
    - url: https://registry.npmjs.org
    - url: https://private.registry.company.org
      scope: "@company"
      authToken: secretToken
```

---

#### `url`

Specifies the URL of the npm registry.

<p><small>| REQUIRED | STRING |</small></p>

```yaml
    - url: https://registry.npmjs.org
```

---

#### `scope`

Specifies which scope is associated with this registry.
See [Associating a scope with a registry](https://docs.npmjs.com/cli/v9/using-npm/scope#associating-a-scope-with-a-registry).

<p><small>| OPTIONAL | STRING |</small></p>

```yaml
    - url: https://registry.npmjs.org
      scope: "@company"
```

---

#### `authToken`

> Specifies the authentitcation token to be used with this registry.

Reviewer 2: ```suggestion
Specifies the authentication token to be used with this registry.
```
